src/dht.cpp

```diff
@@ -209,37 +219,37 @@ struct Dht::SearchNode {
      */
     bool canGet(time_point now, time_point update, std::shared_ptr<Query> q = {}) const {
         /* find request status for the given query */
-        const auto& get_status = q ? getStatus.find(q) : getStatus.cend();
+        const auto& get_status = getStatus.find(q);
         /* find request status for a query satisfying the initial query */
-        const auto& sq_status = not q ? getStatus.cend() :
-            std::find_if(getStatus.cbegin(), getStatus.cend(),
-                [&q](const SyncStatus::value_type& s) {
-                    return s.first and q->isSatisfiedBy(*s.first);
-                }
+        const auto& sq_status = std::find_if(getStatus.cbegin(), getStatus.cend(),
+            [&q](const SyncStatus::value_type& s) {
+                return s.first and q and q->isSatisfiedBy(*s.first);
+            }
         );
         return not node->isExpired() and (now > last_get_reply + Node::NODE_EXPIRE_TIME or update > last_get_reply)
-               and ((get_status == getStatus.cend() or not get_status->second)
-                   and (sq_status == getStatus.cend() or not sq_status->second or not sq_status->second->pending()));
+            and not pending(getStatus)
```

Comment:
Actually, this line could block some requests that could be sent concurrently. The solution is to use `sq_status`. I did have some trouble with it, so I removed it, but I see it's the logical thing to do.